### PR TITLE
MNG - 98 Add useMango to gameHandler class

### PR DESCRIPTION
**Pull request by AgataLudwiczynska**

### CODE REVIEW

File: src/playerClass.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export class Player {
   private _name: string;
   private _pokemons: Pokemon[];
   private _indexOfActivePokemon: number;
+  private _mango: Number;
```
Comment (lukaszdutka) [inline src/playerClass.ts]: There is only one mango per player, so it can be boolean called `_hasMango`

File: src/playerClass.ts [hunk 2]
```diff
@@ -14,6 +15,7 @@ export class Player {
     this._name = name;
     this._pokemons = pokemons;
     this._indexOfActivePokemon = 0;
+    this._mango=1;
```
Comment (lukaszdutka) [inline src/playerClass.ts]: =true

File: src/playerClass.ts [hunk 3]
```diff
@@ -37,6 +39,14 @@ export class Player {
     return alivePokemonsTab;
   }
 
+  reduceNumberMango(): void {
+    this._mango = <any>this._mango - 1;
```
Comment (lukaszdutka) [inline src/playerClass.ts]: = false

File: src/gameHandlerClass.ts [hunk 4]
```diff
@@ -35,4 +35,11 @@ export class GameHandler {
       ? this._playerTwo
       : this._playerOne;
   }
-}
+
+  useMango(): void | Error{
```
Comment (lukaszdutka) [inline src/gameHandlerClass.ts]: ideally there should be one more method: "can I use mango" or something like that, that would be used to render if "use mango" button will be disabled or enabled :)

**Review by lukaszdutka — CHANGES_REQUESTED** [review]
Looks good, but need some polishing :)

File: src/gameHandlerClass.ts [hunk 5]
```diff
@@ -35,4 +35,15 @@ export class GameHandler {
       ? this._playerTwo
       : this._playerOne;
   }
+
+  canIUseMango(): Boolean {
```
Comment (lukaszdutka) [inline src/gameHandlerClass.ts]: **canUseMango** is enough :)

File: src/gameHandlerClass.ts [hunk 6]
```diff
@@ -35,4 +35,15 @@ export class GameHandler {
...
+    return this.getCurrentPlayer().hasMango;
+  }
+
+  useMango(): void | Error {
+    if (this.canIUseMango() == false)
```
Comment (lukaszdutka) [inline src/gameHandlerClass.ts]: .

Comment (lukaszdutka) [reply src/gameHandlerClass.ts]: `this.canIUseMango() == false` it's the same as:
`!this.currentPlayer.hasMango`

File: src/playerClass.ts [hunk 7]
```diff
@@ -37,6 +39,14 @@ export class Player {
     return alivePokemonsTab;
   }
 
+  reduceNumberMango(): void {
```
Comment (lukaszdutka) [inline src/playerClass.ts]: please rename this method :P and add validation inside:
if you ate mango once, then calling this method again should throw an error

File: src/gameHandlerClass.ts (hunk 5 above)
Comment (lukaszdutka) [reply src/gameHandlerClass.ts]: well, this whole method is probably not necessary :P 
But it might be, if it would be used somewhere outside of gamehandler 🤔

Comment (AgataLudwiczynska) [reply src/gameHandlerClass.ts]: I think it will actually be needed to disable button "use Mango"

File: src/gameHandlerClass.ts [hunk 8]
```diff
@@ -35,4 +35,15 @@ export class GameHandler {
       ? this._playerTwo
       : this._playerOne;
   }
+
+  canUseMango(): Boolean {
```
Comment (AleksandraCyp) [inline src/gameHandlerClass.ts]: Why did you add this to game handler? Wee need it only in our player class (we shouldn't put it here because it's a poperty and a method of a player, one player may have a mango, but the second one not necessarily). The whole logic about using mango and healing should be put in the useMango function in playerClass

File: src/playerClass.ts [hunk 9]
```diff
@@ -37,6 +39,16 @@ export class Player {
     return alivePokemonsTab;
   }
 
+  useMango(): void | Error {
```
Comment (AleksandraCyp) [inline src/playerClass.ts:42]: add the whole logic about using mango here, in player class

**Review by AleksandraCyp — CHANGES_REQUESTED** [review]

**Review by AleksandraCyp — APPROVED** [review]
